nixos/modules/hardware/iosched.nix

```diff
@@ -0,0 +1,150 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.hardware.block;
+  escape = lib.strings.escape [ ''"'' ];
+  udevValue = types.addCheck types.nonEmptyStr (x: builtins.match "[^\n\r]*" x != null) // {
+    name = "udevValue";
+    description = "udev rule value";
+    descriptionClass = "noun";
+  };
```

Comment:
👌

This could perhaps be made available as a type or something as this should be something that is generally useful in other modules/user config.

Reply:
Do you think it might be useful to have a generic framework for representing udev rules in Nix?

Reply:
That sounds like a good idea to me, though not necessarily a very high priority.